### PR TITLE
Chore: Use x/xerrors instead of pkg/errors

### PR DESCRIPTION
**Pull request by aocenas**
As discussed at backend weekly we want to use x/xerrors package as that should be in standard library from 1.13. This also removed pkg/errors which was used on some places and has the same functionality.

TODO:
- I wonder if it would make sense to alias xerrors to errors so it is easier to switch later for builtin errors.
- The wrapping with Errorf is kinda too implicit for my taste as you need to use Errorf with `: %w` for error to be wrapped:
```go
package main

import (
	"fmt"
	"golang.org/x/xerrors"
)

var sentinel = xerrors.New("sentinel")

func main() {
	e := wrapping()
	// %+v prints stack trace whether it was wrapped or not but only from point the error was created by xerrors
	// or xerrors.Errorf was used
	fmt.Printf("got error: %+v \n", e)
	// This is true only if %w is used to wrap the error in Errorf
	fmt.Printf("is sentinel: %v", xerrors.Is(e, sentinel))
}

func err() error {
	return sentinel
}

func wrapping() error {
	e := err()
	// This wraps the error because %w
	return xerrors.Errorf("wrapped: %w", e)
	// This does not wrap the error
	//return xerrors.Errorf("wrapped: %v", e)
}
```
So I wonder if it would make sense to use some util `Wrap` function that would make it a bit more explicit.

### CODE REVIEW

**Comment by bergquist** [comment]
That's kinda disappointing. Choosing between %v and %w seems to be error prune. 
I wonder why. 

Yea, in this case, I think aliased import makes sense. 
But I'm not as convinced we should use this anymore. Should we just stay with `pkg/errors` until 1.13?

**Comment by aocenas** [comment]
@bergquist Well if it is going to be standard I am not sure if it makes sense to ignore it especially as `pkg/errors` and this one has almost the same functionality (just this one API is different). Would it make sense to create our own `Wrap` function for it that has the same API as `pkg/errors`? It should be as easy as:
```go
func Wrap(message string, err error) error {
  return xerrors.Errorf("%v: %w", message,  err)
}
```

**Comment by bergquist** [comment]
Yea I think your right. 

I just... dislike importing that all over the codebase.
Seems like we are not the only ones who miss `Wrap` https://github.com/golang/go/issues/29934 :)

**Comment by bergquist** [comment]
Let's add a wrapper function for now. I'm not sure where it should be added but I guess `util` for now.

File: pkg/util/errors.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package util
```
Comment (bergquist) [inline pkg/util/errors.go]: how about we move this into an `errutil` package in `pkg/util/errutil`?

I think its easier to understand
```go
return errutil.Wrap("failed to check if dashboard is provisioned", err)
```

**Review by bergquist — APPROVED** [review]
LGTM

**Pull request merged**